### PR TITLE
Cleanup OC_DB methods

### PR DESCRIPTION
**Pull request by MorrisJobke**
cc @LukasReschke @nickvergessen @PVince81 @rullzer @Xenopathic @DeepDiver1975 

This mostly replaces the method calls with the encapsulated code.

It's quite easy to review if it is done per commit.

### CODE REVIEW

File: lib/private/share/share.php [hunk 1]
```diff
@@ -436,7 +436,7 @@ public static function getItemSharedWithUser($itemType, $itemSource, $user, $own
 
 				// TODO: inject connection, hopefully one day in the future when this
 				// class isn't static anymore...
-				$conn = \OC_DB::getConnection();
+				$conn = \OC::$server->getDatabaseConnection();
```
Comment (nickvergessen) [inline lib/private/share/share.php]: cc @rullzer I hope this doesn't conflict with your changes?

**Comment by nickvergessen** [comment]
Looks good :+1:

File: lib/private/share/share.php (hunk 1 above)
Comment (rullzer) [inline lib/private/share/share.php]: Nah fne with me... I'm not touch share.php ;) That will just be `rm share.php` ;)

Comment (nickvergessen) [inline lib/private/share/share.php]: okay then

**Comment by nickvergessen** [comment]
```
PHP Fatal error:  Call to undefined method OC_DB::getConnection() in /ssd/jenkins/workspace/core-ci-linux@4/database/sqlite/label/SLAVE/tests/lib/app.php on line 464
13:11:50 PHP Stack trace:
13:11:50 PHP   1. {main}() /usr/local/bin/phpunit:0
13:11:50 PHP   2. PHPUnit_TextUI_Command::main() /usr/local/bin/phpunit:537
13:11:50 PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:105
13:11:50 PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:153
13:11:50 PHP   5. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php:406
13:11:50 PHP   6. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
13:11:50 PHP   7. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
13:11:50 PHP   8. PHPUnit_Framework_TestCase->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
13:11:50 PHP   9. PHPUnit_Framework_TestResult->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:699
13:11:50 PHP  10. PHP_Invoker->invoke() phar:///usr/local/bin/phpunit/phpunit/Framework/TestResult.php:607
13:11:50 PHP  11. call_user_func_array:{phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93}() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
13:11:50 PHP  12. PHPUnit_Framework_TestCase->runBare() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
13:11:50 PHP  13. PHPUnit_Framework_TestCase->runTest() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:743
13:11:50 PHP  14. ReflectionMethod->invokeArgs() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:866
13:11:50 PHP  15. Test_App->testEnabledApps() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:866
13:11:50 PHP  16. Test_App->setupAppConfigMock() /ssd/jenkins/workspace/core-ci-linux@4/database/sqlite/label/SLAVE/tests/lib/app.php:397
13:11:57 Build step 'Execute shell' marked build as failure
13:11:57 Recording test results
13:12:01 Setting commit status on GitHub for https://github.com/owncloud/core/commit/5470a633f904f6bb3d851baf9203d9338aaebd16
13:12:05 Finished: FAILURE
```

**Comment by MorrisJobke** [comment]
@nickvergessen Fixed. It was only this single occurrence that was left untouched :see_no_evil:

**Comment by rullzer** [comment]
\o/ cleanups!
:+1:

**Comment by DeepDiver1975** [comment]
@MorrisJobke rebase please - THX

**Comment by MorrisJobke** [comment]
> @MorrisJobke rebase please - THX

Done.

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**